codegen-lite/src/test/resources/software/amazon/awssdk/codegen/lite/defaultsmode/defaults-mode-configuration.java

```diff
@@ -0,0 +1,80 @@
+                                                                    .put(SdkClientOption.DEFAULT_RETRY_MODE, RetryMode.ADAPTIVE).build();
+
+    private static final AttributeMap MOBILE_HTTP_DEFAULTS = AttributeMap.builder()
+                                                                         .put(SdkHttpConfigurationOption.CONNECTION_TIMEOUT, Duration.ofMillis(10000)).build();
+
+    private static final AttributeMap CROSS_REGION_DEFAULTS = AttributeMap.builder()
+                                                                          .put(SdkClientOption.DEFAULT_RETRY_MODE, RetryMode.STANDARD).build();
+
+    private static final AttributeMap CROSS_REGION_HTTP_DEFAULTS = AttributeMap.builder()
+                                                                               .put(SdkHttpConfigurationOption.CONNECTION_TIMEOUT, Duration.ofMillis(2800)).build();
+
+    private static final AttributeMap IN_REGION_DEFAULTS = AttributeMap.builder()
+                                                                       .put(SdkClientOption.DEFAULT_RETRY_MODE, RetryMode.STANDARD).build();
+
+    private static final AttributeMap IN_REGION_HTTP_DEFAULTS = AttributeMap.builder()
+                                                                            .put(SdkHttpConfigurationOption.CONNECTION_TIMEOUT, Duration.ofMillis(1000)).build();
+
+    private static final AttributeMap LEGACY_DEFAULTS = AttributeMap.empty();
+
+    private static final AttributeMap LEGACY_HTTP_DEFAULTS = AttributeMap.empty();
+
+    private static final Map<DefaultsMode, AttributeMap> DEFAULT_CONFIG_BY_MODE = new EnumMap<>(DefaultsMode.class);
+
+    private static final Map<DefaultsMode, AttributeMap> DEFAULT_HTTP_CONFIG_BY_MODE = new EnumMap<>(DefaultsMode.class);
+
+    static {
+        DEFAULT_CONFIG_BY_MODE.put(DefaultsMode.STANDARD, STANDARD_DEFAULTS);
+        DEFAULT_CONFIG_BY_MODE.put(DefaultsMode.MOBILE, MOBILE_DEFAULTS);
+        DEFAULT_CONFIG_BY_MODE.put(DefaultsMode.CROSS_REGION, CROSS_REGION_DEFAULTS);
+        DEFAULT_CONFIG_BY_MODE.put(DefaultsMode.IN_REGION, IN_REGION_DEFAULTS);
+        DEFAULT_CONFIG_BY_MODE.put(DefaultsMode.LEGACY, LEGACY_DEFAULTS);
+        DEFAULT_HTTP_CONFIG_BY_MODE.put(DefaultsMode.STANDARD, STANDARD_HTTP_DEFAULTS);
+        DEFAULT_HTTP_CONFIG_BY_MODE.put(DefaultsMode.MOBILE, MOBILE_HTTP_DEFAULTS);
+        DEFAULT_HTTP_CONFIG_BY_MODE.put(DefaultsMode.CROSS_REGION, CROSS_REGION_HTTP_DEFAULTS);
+        DEFAULT_HTTP_CONFIG_BY_MODE.put(DefaultsMode.IN_REGION, IN_REGION_HTTP_DEFAULTS);
+        DEFAULT_HTTP_CONFIG_BY_MODE.put(DefaultsMode.LEGACY, LEGACY_HTTP_DEFAULTS);
+    }
+
+    private DefaultsModeConfiguration() {
+    }
+
+    /**
+     * Return the default config options for a given defaults mode
+     */
+    public static AttributeMap defaultConfig(DefaultsMode mode) {
+        return DEFAULT_CONFIG_BY_MODE.getOrDefault(mode, AttributeMap.empty());
```

Comment:
Unfortunately it looks like `AttributeMap.empty()` returns a new instance, rather than a singleton. To minimize allocations, we should either make `AttributeMap.empty()` a singleton, or only call `empty()` when the key is not present.

Reply:
Good call. Will create a singleton for AttributeMap.empty()